save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/TopBar.kt

```diff
@@ -54,6 +54,148 @@ external interface TopBarProps : PropsWithChildren {
+            SituationUrlClassification.DETAILS
+        } else if (href.contains(Regex("/[^/]+/[^/]+/history/execution"))) {
+            SituationUrlClassification.EXECUTION
+        } else {
+            SituationUrlClassification.KEYWORD_PROCESS
+        }
+    }
+
+    /** Function check exclude and generate currentPath after the buttons creating
+     *
+     * @param pathPart
+     */
+    private fun fixCurrentPathAfter(pathPart: String) {
+        currentPath = when (circumstance) {
+            SituationUrlClassification.PROJECT_OR_ORGANIZATION, SituationUrlClassification.ARCHIVE -> "#"
+            SituationUrlClassification.DETAILS, SituationUrlClassification.EXECUTION -> if (pathPart == "execution") mergeUrls(pathPart) else currentPath
+            else -> currentPath
+        }
+    }
+
+    /** The function changes the exclude after the button is created
+     *
+     * @param isNumber
+     */
+    private fun fixExcludeAfter(isNumber: Boolean) {
+        circumstance = when (circumstance) {
+            SituationUrlClassification.PROJECT_OR_ORGANIZATION, SituationUrlClassification.ARCHIVE -> SituationUrlClassification.KEYWORD_PROCESS
+            SituationUrlClassification.DETAILS -> if (isNumber) setProcessLastSegments(1) else SituationUrlClassification.DETAILS
+            else -> circumstance
+        }
+    }
+
+    private fun mergeUrls(secondPath: String) = "$currentPath/$secondPath"
+
+    private fun setProcessLastSegments(number: Int): SituationUrlClassification {
+        processLastSegments = number
+        return SituationUrlClassification.KEYWORD_PROCESS_LAST_SEGMENTS
+    }
+
+    /**
+     * This Enum class classifies work with the url address segment
+     */
+    enum class SituationUrlClassification {
+        /**
+         * ARCHIVE - situation with the processing of the "archive" in the url address - need for tabs in AwesomeBenchmarksView
+         */
+        ARCHIVE,
+
+        /**
+         * DETAILS - situation with the processing of the "details" in the url address - need for deleted multi-segment urls, starting with the word "details"
+         */
+        DETAILS,
+
+        /**
+         * EXECUTION - situation with the processing of the "execution" in the url address - need for redirect to the page with the executions history
+         */
+        EXECUTION,
+
+        /**
+         * KEYWORD_NOT_PROCESS - the button with this url segment is not created
+         */
+        KEYWORD_NOT_PROCESS,
+
+        /**
+         * KEYWORD_PROCESS - the button with this url segment is created without changes
+         */
+        KEYWORD_PROCESS,
+
+        /**
+         * KEYWORD_PROCESS_LAST_SEGMENTS - a button is created if this segment is one of the last
+         */
+        KEYWORD_PROCESS_LAST_SEGMENTS,
+
+        /**
+         * PROJECT_OR_ORGANIZATION - situation with the processing of the "archive" in the url address - need for tabs in OrganizationView and ProjectView
```

Comment:
No need to put name of enum entity into kdoc

Reply:
Ok, up to you

Reply:
Done